lib/mayaUsd/python/wrapPrimWriter.cpp

```diff
@@ -206,20 +206,36 @@ class PrimWriterWrapper
             return sptr;
         }
 
+        UsdMayaPrimWriter::ContextSupport
+        operator()(const UsdMayaJobExportArgs& exportArgs, const MObject& exportObj)
+        {
+
+            boost::python::object pyClass = GetPythonObject(_classIndex);
+            if (!pyClass) {
+                // Prototype was unregistered
+                return UsdMayaPrimWriter::ContextSupport::Unsupported;
+            }
+            TfPyLock pyLock;
+            if (PyObject_HasAttrString(pyClass.ptr(), "CanExport")) {
+                boost::python::object CanExport = pyClass.attr("CanExport");
+                PyObject*             callable = CanExport.ptr();
+                auto res = boost::python::call<int>(callable, exportArgs, exportObj);
+                return UsdMayaPrimWriter::ContextSupport(res);
+            } else {
+                return UsdMayaPrimWriter::ContextSupport::Fallback;
+            }
+        }
+
         // Create a new wrapper for a Python class that is seen for the first time for a given
         // purpose. If we already have a registration for this purpose: update the class to
         // allow the previously issued factory function to use it.
-        static UsdMayaPrimWriterRegistry::WriterFactoryFn
-        Register(boost::python::object cl, const std::string& mayaTypeName)
+        static FactoryFnWrapper
+        Register(boost::python::object cl, const std::string& mayaTypeName, bool& updated)
         {
             size_t classIndex = RegisterPythonObject(cl, GetKey(cl, mayaTypeName));
-            if (classIndex != UsdMayaPythonObjectRegistry::UPDATED) {
-                // Return a new factory function:
-                return FactoryFnWrapper { classIndex };
-            } else {
-                // We already registered a factory function for this purpose:
-                return nullptr;
-            }
+            updated = classIndex == UsdMayaPythonObjectRegistry::UPDATED;
```

Comment:
I'm never sure about = vs == priority, so as a dev reading the code, parenthesis would make me feel safer.